src/components/Icon.js

```diff
@@ -1,15 +1,28 @@
 import PropTypes from 'prop-types';
-import React from 'react';
-import { View } from 'react-native';
+import React, { useContext } from 'react';
+
+import { StyleSheet, View } from 'react-native';
 import { SvgXml } from 'react-native-svg';
 
 import { normalize } from '../config';
+import { OrientationContext } from '../OrientationProvider';
+
+export const Icon = ({ icon, width, height, style }) => {
+  const { orientation } = useContext(OrientationContext);
+  const landscapeStyle = orientation === 'landscape' ? style : null;
```

Comment:
using `style` for just for landscape changes the whole logic.
the style was there before in every case. now it is just applied to the container `View`, if in landscape mode.

this will lead to problems in components, where `style` is already used, for example in `Link`:

```
<Icon icon={link(colors.secondary)} style={styles.icon} />
```

that icon styles now only apply on landscape mode.

what you want is a new `landscapeStyle`, which is not related to the main style, which can be passed. it does not need to be a component prop i think, or do we need to pass landscape style from outside? maybe to have the ability to customize it. ok, so `landscapeStyle` can be added in the props:

```
({ icon, width, height, style, landscapeStyle })
```

in `View` than both styles need to be added with an array, while the landscape style only, if orientation is landscape:

```
<View style={[style, orientation === 'landscape' && landscapeStyle]}>
```

and as defaultProps you need to change `style`, which should have no default, to `landscapeStyle`.

have you understand? otherwise reach me on Slack.

Reply:
all clear , I'm going to take care of it now .